scripts/cleanup_analysis_entries.py

```diff
@@ -0,0 +1,259 @@
+    # get all analysis-entries 
+    return aapi.query_analyses(
+            AnalysisQueryModel(
+                projects=datasets,
+                type=AnalysisType(analysis_type),
+                # status=AnalysisStatus('completed'),
+            )
+        )
+
+def find_invalid_analyses(analyses: list[dict]):
+    """
+    
+    
+    """
+    # flag analyses in a bad state
+    pass
+
+
+def get_path_components_from_path(path):
+    """
+    Return the {bucket_name}, {dataset}, and {subdir} for GS only paths
+
+    >>> get_bucket_name_path_from_path('gs://cpg-dataset-main/subfolder/subfolder2/my.cram')
+    {bucket_name:'cpg-dataset-main', dataset:'dataset', subdir:'subfolder/subfolder2'}
+    """
+    if path.startswith('gs://'):
+        short_path = path.removeprefix('gs://').split('/', maxsplit=1)
+        bucket_name = short_path[0]
+
+        dataset_with_bucket_type = bucket_name.removeprefix('cpg-')
+        bucket_type = dataset_with_bucket_type.split('-')[-1]
+        dataset = dataset_with_bucket_type.removesuffix(bucket_type)[:-1]
+
+        file_name = short_path[1].split('/')[-1]
+        subdir = short_path[1].removesuffix(file_name)[:-1]
+
+    else:
+        raise warnings.warn(f'Analysis path {path} not bucket storage.')
+    
+    return {'bucket_name':bucket_name, 'dataset':dataset, 'subdir':subdir}
+
+def deduplicate_analyses(analyses: list[dict]) -> tuple[list[dict], list[int], list[list]]:
+    # Check which analysis IDs point to the same analysis output path
+    # Keep the smallest ID pointing to a given analysis output, duplicates can be scheduled for deletion
+
+    all_analyses = defaultdict(str)
+    # map each analysis id to its output path
+    for analysis in analyses:
+        all_analyses[analysis['id']]=analysis['output']
+    
+    # map each output path to the analysis entries which point to it
+    reversed_analyses_dict = {}
+    for k, v in all_analyses.items():
+        reversed_analyses_dict.setdefault(v, set()).add(k)
+
+    duplicate_analyses = [vals for vals in reversed_analyses_dict.values() if len(vals)>1]
+    
+    # keep the first entry pointing to a path, duplicates can be deleted
+    duplicate_analyses_ids = []
+    for duplicate_ids in duplicate_analyses:
+        duplicate_analyses_ids.append(sorted(duplicate_ids)[1:])
+
+    ids_to_delete = [e for sublist in duplicate_analyses_ids for e in sublist]
+
+    return (analyses, ids_to_delete, duplicate_analyses_ids)
+
+def check_all_analyses_belongs_to_project(analyses: list[dict], project_id_map: dict):
+    # Check the analyses ['project'] matches the bucket name, `cpg-{dataset}-{main/test}`
+
+    # bad_mappings = [] # Use logger for this
+
+    for analysis in analyses:
+        project_id = analysis['project']
+        project_name = project_id_map[project_id]
+        dataset_name = get_path_components_from_path(analysis['output'])['dataset']
+        if dataset_name == project_name:
+            continue
+        else:
```

Comment:
Don't need the `else` here as the if: continue early returns (love early returns)